enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationService.java

```diff
@@ -534,12 +534,12 @@ public List<DocumentMetadataResponseDto> createDocsMetadata(List<DocumentVerific
         entities.forEach(entity -> {
             DocumentMetadataResponseDto docMetadata = toDocumentMetadata(entity);
 
-            if (DocumentStatus.REJECTED.equals(entity.getStatus())) {
-                List<String> errors = collectRejectionErrors(entity);
-                if (docMetadata.getErrors() == null) {
-                    docMetadata.setErrors(new ArrayList<>());
+            if (DocumentStatus.REJECTED == entity.getStatus()) {
+                final List<String> errors = collectRejectionErrors(entity);
```

Comment:
Just thinking. Does it make sense to return from `collectRejectionErrors`? Should not we refactor it into something like `auditRejectionErrors`?

Reply:
We can do that. The `collectRejectionErrors` is called only in this case and so it is sufficient to return boolean from the method, indicating whether an error exists for a document.

Reply:
Actually we do not need to call the method `collectRejectionErrors` at all. It is quite complex function that visits database twice and does deserialisation, just for getting a messages that are not used anyway.

1. Result of a verification is logged and stored right after the verification process is finished by a document verification provider. So we won't lose any information about a document.
2. A document that is failed has error detail, a document that is rejected has rejection reason for sure.
3. The method `collectRejectionErrors` did not set any other attribute than the errors, which should not leak any details now.
4. The method is called within `/document/status` endpoint and we do not want to include specific error details anymore. Using this method just to log rejection details does not seem as expected behaviour to me.
5. The audit function only logs access to the document provider.